_posts/2019-04-17-sod.md

```diff
@@ -0,0 +1,54 @@
+
+## Project Ideas List
+
+### Improve Apache Flink Stream Processing Concepts Documentation
+
+**Description:** Stream processing is the processing of data in motion―in other words, computing on data directly as it is produced or received. Apache Flink has pioneered the field of distributed, stateful stream processing for the last few years. As such the community has introduced a couple of new concepts, which you need to become familiar with to develop and operate stream processing applications with Apache Flink efficiently.
+The Apache Flink documentation \[1\] already contains a “concepts” section, but it is a ) incomplete and b) lacks an overall structure & reading flow. In addition, “concepts”-content is also spread over the development \[2\] & operations \[3\] documentation without references to the “concepts” section. An example of  this can be found in \[4\] and  \[5\].
+In this project, we would like to restructure, consolidate and extend the concepts documentation for Apache Flink. This documentation should also be linked to & from the deployment and development documentation to guide users, who want to become productive as quickly as possible.
+
+**Related material:**
+
+1. https://ci.apache.org/projects/flink/flink-docs-release-1.8/
+2. https://ci.apache.org/projects/flink/flink-docs-release-1.8/dev
+3. https://ci.apache.org/projects/flink/flink-docs-release-1.8/ops/
+4. https://ci.apache.org/projects/flink/flink-docs-release-1.8/concepts/programming-model.html#time
+5. https://ci.apache.org/projects/flink/flink-docs-release-1.8/dev/event_time.html
+
+### Improve Apache Flink Deployment & Operations Documentation
+
+**Description:** Stream processing is the processing of data in motion―in other words, computing on data directly as it is produced or received. Apache Flink has pioneered the field of distributed, stateful stream processing for the last few years. As a stateful distributed system in general and a continuously running, low-latency system in particular, Apache Flink deployments are not trivial to setup and manage.
+Unfortunately, the operations \[1\] as well as the monitoring documentation \[2\] are arguably the weakest spots of the Apache Flink documentation. While it is comprehensive and often goes into a lot of detail, it lacks an overall structure and does not address common overarching concerns of operations teams in an efficient way.
+In this project, we would like to restructure this part of the documentation, properly link it to the relevant stream processing concepts (see above) and extend it if possible. Ideas for extension include: discussion of session and per-job clusters, better documentation for containerized deployments (incl. K8s), capacity planning &  integration into CI/CD pipelines.
```

Comment:
The description sounds as if project 1 is a prerequisite of project 2 (`link it to relevant stream processing concepts (see above)`. IMO, it would be better to have independent projects.

Reply:
Does it just sounds like it or do you think they do build on each other?

Reply:
I think we can describe the project in a way that it they are independent, just putting different emphasis. In case both are accepted, they can fit together but I think there are many valuable improvements that do not relate to the first project.

Reply:
What about dropping the references to linking sections? It sounds more like an implementation detail of the project vs a core selling point.

Reply:
I agree. I simply removed the reference to the stream processing concepts. I don't think this is necessary to understand the project.